### ISSUE TITLE
Menu API

### ISSUE THREAD

**Issue by emoon** [emoon/ProDBG#90]
Figure out how to deal with menus in ProDBG.

I would like to have a ImGui style API because it's nice to use.
Still want the possibility to support "native" menus when wanted. For example on Mac the title menu is very common in all apps so I think it would be better to use that compared to for example a ImGui one there while on Linux (and Windows perhaps) using the ImGui style one would be ok.

**Comment by theWatchmen** [comment]
Related to this, I have started implementing contextual menu entries for the popup menu. Still need to cleanup the code, but if you want to have a look it's on the imgui-menus branch.

**Comment by emoon** [comment]
Yeah I just noticed you committed there :) Also this one is just moved over from the phoreplay tracker. I will move back all stuff here again. Github supports what I want to use now and it just makes it easier for people in general if the issues are here.

**Comment by emoon** [comment]
Quite a bit has changed since I opened this one. With Rust coming it I have now done some work on this already. Popup menus now always uses Imgui while "real" menus uses System ones on Windows and Mac. On Linux we should use Imgui for it though which needs to be implemented. Currently Window handling is being done in the https://github.com/emoon/rust_minifb which ProDBG uses.
